module/irrecoverable/exception.go

```diff
@@ -0,0 +1,43 @@
+package irrecoverable
+
+import (
+	"fmt"
+)
+
+// exception represents an unexpected error. An unexpected error is any error returned
+// by a function, other than the error specifically documented as expected in that
+// function's interface.
+//
+// It wraps an error, which could be a sentinel error. IT does NOT IMPLEMENT an UNWRAP method,
```

Comment:
This means, we never need to handle this exception error. My understanding is that, the reason we would use this exception error type is to prevent the inner wrapped error from being accessed by `errors.As` and `errors.Is`, which might cause the exception error to be handled as a non-exception sentinel error?

Reply:
> This means, we never need to handle this exception error.
Exactly. Any time we interpret some error as unexpected, even if that error is a sentinel, it should not later be interpetable as a sentinel. 

If it is desirable that the error should later be interpreted as a sentinel, then it cannot be interpreted as unexpected at the earlier level in the callstack. 

This is a tool to assert that an error is an exception, and to avoid bugs where an unexpected error is inadvertently interpreted as an expected sentinel, because of Go's error unwrapping. 

There is an example of the situation this is trying to avoid here: https://github.com/onflow/flow-go/pull/3903#discussion_r1113512979